### ISSUE TITLE
Panics for HTTP2 requests that close the connection before reading the body (if streamed)

### ISSUE THREAD

**Issue by rolftimmermans** [hyperium/hyper#1620]
If a body is streamed with Hyper, then a client that closes the connection before closing it seems to cause panics.

Here is the backtrace I receive:

```
thread 'tokio-runtime-worker-1' panicked at 'state=State { inner: Closed(Io) }', /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/h2-0.1.11/src/proto/streams/prioritize.rs:391:9
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: <std::panicking::begin_panic::PanicPayload<A> as core::panic::BoxMeUp>::get
             at libstd/panicking.rs:511
   5: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:426
   6: std::panicking::try::do_call
             at libstd/panicking.rs:413
   7: h2::proto::streams::prioritize::Prioritize::assign_connection_capacity::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/h2-0.1.11/src/proto/streams/prioritize.rs:391
   8: h2::proto::streams::prioritize::Prioritize::new
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/h2-0.1.11/src/proto/streams/prioritize.rs:256
   9: <http::header::value::HeaderValue as core::convert::AsRef<[u8]>>::as_ref
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/h2-0.1.11/src/proto/streams/send.rs:245
  10: <h2::proto::streams::streams::StreamRef<B>>::reserve_capacity
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/h2-0.1.11/src/proto/streams/streams.rs:910
  11: <h2::share::SendStream<B>>::reserve_capacity
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/h2-0.1.11/src/share.rs:266
  12: <hyper::proto::h2::PipeToSendStream<S> as futures::future::Future>::poll
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.12.7/src/proto/h2/mod.rs:109
  13: <hyper::proto::h2::server::H2Stream<F, B>>::poll2
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.12.7/src/proto/h2/server.rs:220
  14: <hyper::proto::h2::server::H2Stream<F, B> as futures::future::Future>::poll
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.12.7/src/proto/h2/server.rs:238
  15: alloc::alloc::realloc
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/future/mod.rs:113
  16: <bool as core::fmt::Debug>::fmt
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/task_impl/mod.rs:289
  17: <bool as core::fmt::Debug>::fmt
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/task_impl/mod.rs:363
  18: futures::task_impl::std::CURRENT_THREAD_NOTIFY::__init
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/task_impl/std/mod.rs:78
  19: <bool as core::fmt::Debug>::fmt
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/task_impl/mod.rs:363
  20: <bool as core::fmt::Debug>::fmt
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.23/src/task_impl/mod.rs:289
  21: tokio_threadpool::task::TaskFuture::poll
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/task/mod.rs:292
  22: tokio_threadpool::task::Task::run::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/task/mod.rs:165
  23: core::ops::function::FnOnce::call_once
             at /Users/travis/build/rust-lang/rust/src/libcore/ops/function.rs:223
  24: <tokio_threadpool::worker::stack::State as core::cmp::PartialEq>::ne
             at /Users/travis/build/rust-lang/rust/src/libstd/panic.rs:308
  25: crossbeam_epoch::epoch::Epoch::starting
             at /Users/travis/build/rust-lang/rust/src/libstd/panicking.rs:310
  26: panic_unwind::dwarf::eh::read_encoded_pointer
             at libpanic_unwind/lib.rs:105
  27: crossbeam_epoch::epoch::Epoch::starting
             at /Users/travis/build/rust-lang/rust/src/libstd/panicking.rs:289
  28: crossbeam_epoch::guard::unprotected
             at /Users/travis/build/rust-lang/rust/src/libstd/panic.rs:392
  29: tokio_threadpool::task::Task::run
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/task/mod.rs:151
  30: tokio_threadpool::worker::Worker::run_task2
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:545
  31: tokio_threadpool::worker::Worker::run_task
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:445
  32: tokio_threadpool::worker::Worker::try_steal_task
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:412
  33: tokio_threadpool::worker::Worker::try_run_task
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:297
  34: tokio_threadpool::worker::Worker::with_current::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:239
  35: tokio::runtime::builder::Builder::build::{{closure}}::{{closure}}::{{closure}}::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.7/src/runtime/builder.rs:125
  36: tokio_timer::clock::clock::CLOCK::__init
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.4/src/timer/handle.rs:64
  37: std::sync::mpsc::blocking::SignalToken::cast_from_usize
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:294
  38: std::sync::mpsc::blocking::SignalToken::cast_from_usize
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:248
  39: tokio_timer::clock::clock::CLOCK::__init
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.4/src/timer/handle.rs:56
  40: tokio::runtime::builder::Builder::build::{{closure}}::{{closure}}::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.7/src/runtime/builder.rs:124
  41: <std::collections::hash::table::SafeHash as core::cmp::PartialEq>::eq
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.4/src/clock/clock.rs:136
  42: std::sync::mpsc::blocking::SignalToken::cast_from_usize
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:294
  43: std::sync::mpsc::blocking::SignalToken::cast_from_usize
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:248
  44: <std::collections::hash::table::SafeHash as core::cmp::PartialEq>::eq
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.4/src/clock/clock.rs:119
  45: tokio::runtime::builder::Builder::build::{{closure}}::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.7/src/runtime/builder.rs:123
  46: tokio::executor::current_thread::CURRENT::__init
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-reactor-0.1.2/src/lib.rs:231
  47: std::sync::mpsc::blocking::SignalToken::cast_from_usize
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:294
  48: std::sync::mpsc::blocking::SignalToken::cast_from_usize
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:248
  49: tokio::executor::current_thread::CURRENT::__init
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-reactor-0.1.2/src/lib.rs:214
  50: tokio::runtime::builder::Builder::build::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.7/src/runtime/builder.rs:122
  51: tokio_threadpool::callback::Callback::call
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/callback.rs:21
  52: tokio_threadpool::worker::Worker::do_run::{{closure}}::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:121
  53: core::ptr::swap_nonoverlapping_bytes
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-executor-0.1.2/src/global.rs:176
  54: <tokio_threadpool::task::blocking::State as core::cmp::PartialEq>::eq
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:294
  55: <tokio_threadpool::task::blocking::State as core::cmp::PartialEq>::eq
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:248
  56: core::ptr::swap_nonoverlapping_bytes
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-executor-0.1.2/src/global.rs:150
  57: tokio_threadpool::worker::Worker::do_run::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:119
  58: <tokio_threadpool::task::blocking::State as core::cmp::PartialEq>::eq
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:294
  59: <tokio_threadpool::task::blocking::State as core::cmp::PartialEq>::eq
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:248
  60: tokio_threadpool::worker::Worker::do_run
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:110
  61: tokio_threadpool::pool::Pool::spawn_thread::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/pool/mod.rs:417
  62: crossbeam_epoch::collector::Handle::is_pinned
             at /Users/travis/build/rust-lang/rust/src/libstd/sys_common/backtrace.rs:136
  63: tokio_threadpool::task::TaskFuture::poll
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/mod.rs:409
  64: <tokio_threadpool::worker::stack::State as core::cmp::PartialEq>::ne
             at /Users/travis/build/rust-lang/rust/src/libstd/panic.rs:308
  65: crossbeam_epoch::epoch::Epoch::starting
             at /Users/travis/build/rust-lang/rust/src/libstd/panicking.rs:310
  66: panic_unwind::dwarf::eh::read_encoded_pointer
             at libpanic_unwind/lib.rs:105
  67: crossbeam_epoch::epoch::Epoch::starting
             at /Users/travis/build/rust-lang/rust/src/libstd/panicking.rs:289
  68: crossbeam_epoch::guard::unprotected
             at /Users/travis/build/rust-lang/rust/src/libstd/panic.rs:392
  69: tokio_threadpool::task::TaskFuture::poll
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/mod.rs:408
  70: core::fmt::Arguments::new_v1
             at /Users/travis/build/rust-lang/rust/src/liballoc/boxed.rs:640
  71: std::sys_common::thread::start_thread
             at /Users/travis/build/rust-lang/rust/src/liballoc/boxed.rs:650
             at libstd/sys_common/thread.rs:24
  72: std::sys::unix::thread::Thread::new::thread_start
             at libstd/sys/unix/thread.rs:90
  73: _pthread_body
  74: _pthread_start
thread 'tokio-runtime-worker-4' panicked at 'assertion failed: self.slab.is_empty()', /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/h2-0.1.11/src/proto/streams/store.rs:211:13
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: <std::panicking::begin_panic::PanicPayload<A> as core::panic::BoxMeUp>::get
             at libstd/panicking.rs:511
   5: alloc::alloc::dealloc
             at /Users/travis/build/rust-lang/rust/src/libstd/panicking.rs:445
   6: <h2::frame::stream_id::StreamId as core::cmp::PartialEq>::ne
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/h2-0.1.11/src/proto/streams/store.rs:211
   7: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
   8: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
   9: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  10: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  11: alloc::raw_vec::alloc_guard
             at /Users/travis/build/rust-lang/rust/src/liballoc/arc.rs:518
  12: <http::byte_str::ByteStr as core::cmp::PartialEq>::eq
             at /Users/travis/build/rust-lang/rust/src/liballoc/arc.rs:970
  13: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  14: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  15: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  16: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  17: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  18: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  19: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  20: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  21: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  22: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  23: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  24: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  25: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  26: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  27: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  28: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  29: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  30: core::ptr::drop_in_place
             at /Users/travis/build/rust-lang/rust/src/libcore/ptr.rs:59
  31: tokio_threadpool::task::Task::drop_future
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/task/mod.rs:272
  32: tokio_threadpool::task::Task::run
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/task/mod.rs:183
  33: tokio_threadpool::worker::Worker::run_task2
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:545
  34: tokio_threadpool::worker::Worker::run_task
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:445
  35: tokio_threadpool::worker::Worker::try_steal_task
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:412
  36: tokio_threadpool::worker::Worker::try_run_task
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:297
  37: tokio_threadpool::worker::Worker::with_current::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:239
  38: tokio::runtime::builder::Builder::build::{{closure}}::{{closure}}::{{closure}}::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.7/src/runtime/builder.rs:125
  39: tokio_timer::clock::clock::CLOCK::__init
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.4/src/timer/handle.rs:64
  40: std::sync::mpsc::blocking::SignalToken::cast_from_usize
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:294
  41: std::sync::mpsc::blocking::SignalToken::cast_from_usize
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:248
  42: tokio_timer::clock::clock::CLOCK::__init
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.4/src/timer/handle.rs:56
  43: tokio::runtime::builder::Builder::build::{{closure}}::{{closure}}::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.7/src/runtime/builder.rs:124
  44: <std::collections::hash::table::SafeHash as core::cmp::PartialEq>::eq
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.4/src/clock/clock.rs:136
  45: std::sync::mpsc::blocking::SignalToken::cast_from_usize
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:294
  46: std::sync::mpsc::blocking::SignalToken::cast_from_usize
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:248
  47: <std::collections::hash::table::SafeHash as core::cmp::PartialEq>::eq
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.4/src/clock/clock.rs:119
  48: tokio::runtime::builder::Builder::build::{{closure}}::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.7/src/runtime/builder.rs:123
  49: tokio::executor::current_thread::CURRENT::__init
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-reactor-0.1.2/src/lib.rs:231
  50: std::sync::mpsc::blocking::SignalToken::cast_from_usize
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:294
  51: std::sync::mpsc::blocking::SignalToken::cast_from_usize
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:248
  52: tokio::executor::current_thread::CURRENT::__init
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-reactor-0.1.2/src/lib.rs:214
  53: tokio::runtime::builder::Builder::build::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.7/src/runtime/builder.rs:122
  54: tokio_threadpool::callback::Callback::call
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/callback.rs:21
  55: tokio_threadpool::worker::Worker::do_run::{{closure}}::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:121
  56: core::ptr::swap_nonoverlapping_bytes
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-executor-0.1.2/src/global.rs:176
  57: <tokio_threadpool::task::blocking::State as core::cmp::PartialEq>::eq
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:294
  58: <tokio_threadpool::task::blocking::State as core::cmp::PartialEq>::eq
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:248
  59: core::ptr::swap_nonoverlapping_bytes
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-executor-0.1.2/src/global.rs:150
  60: tokio_threadpool::worker::Worker::do_run::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:119
  61: <tokio_threadpool::task::blocking::State as core::cmp::PartialEq>::eq
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:294
  62: <tokio_threadpool::task::blocking::State as core::cmp::PartialEq>::eq
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/local.rs:248
  63: tokio_threadpool::worker::Worker::do_run
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/worker/mod.rs:110
  64: tokio_threadpool::pool::Pool::spawn_thread::{{closure}}
             at /Users/rolftimmermans/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.5/src/pool/mod.rs:417
  65: crossbeam_epoch::collector::Handle::is_pinned
             at /Users/travis/build/rust-lang/rust/src/libstd/sys_common/backtrace.rs:136
  66: tokio_threadpool::task::TaskFuture::poll
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/mod.rs:409
  67: <tokio_threadpool::worker::stack::State as core::cmp::PartialEq>::ne
             at /Users/travis/build/rust-lang/rust/src/libstd/panic.rs:308
  68: crossbeam_epoch::epoch::Epoch::starting
             at /Users/travis/build/rust-lang/rust/src/libstd/panicking.rs:310
  69: panic_unwind::dwarf::eh::read_encoded_pointer
             at libpanic_unwind/lib.rs:105
  70: crossbeam_epoch::epoch::Epoch::starting
             at /Users/travis/build/rust-lang/rust/src/libstd/panicking.rs:289
  71: crossbeam_epoch::guard::unprotected
             at /Users/travis/build/rust-lang/rust/src/libstd/panic.rs:392
  72: tokio_threadpool::task::TaskFuture::poll
             at /Users/travis/build/rust-lang/rust/src/libstd/thread/mod.rs:408
  73: core::fmt::Arguments::new_v1
             at /Users/travis/build/rust-lang/rust/src/liballoc/boxed.rs:640
  74: std::sys_common::thread::start_thread
             at /Users/travis/build/rust-lang/rust/src/liballoc/boxed.rs:650
             at libstd/sys_common/thread.rs:24
  75: std::sys::unix::thread::Thread::new::thread_start
             at libstd/sys/unix/thread.rs:90
  76: _pthread_body
  77: _pthread_start
```

Tested with `curl --http2-prior-knowledge -I -XGET http://localhost...`

`-I` combined with `-XGET` causes a `GET` request to be done rather than `HEAD` request, but still ignores any body that was delivered by the server.

Seems like this should be allowed and the server doesn't have to panic, right?

**Comment by seanmonstar** [comment]
This *sort of* sounds like https://github.com/carllerche/h2/pull/295, which the fix was just published in h2 v0.1.11 this week. Do you know if using that version fixes this?

**Comment by rolftimmermans** [comment]
The above is the output as tested with h2 0.1.11:

```
[[package]]
name = "hyper"
version = "0.12.7"
source = "registry+https://github.com/rust-lang/crates.io-index"
dependencies = [
 "bytes 0.4.9 (registry+https://github.com/rust-lang/crates.io-index)",
 "futures 0.1.23 (registry+https://github.com/rust-lang/crates.io-index)",
 "futures-cpupool 0.1.8 (registry+https://github.com/rust-lang/crates.io-index)",
 "h2 0.1.11 (registry+https://github.com/rust-lang/crates.io-index)",
 "http 0.1.8 (registry+https://github.com/rust-lang/crates.io-index)",
 "httparse 1.3.2 (registry+https://github.com/rust-lang/crates.io-index)",
 "iovec 0.1.2 (registry+https://github.com/rust-lang/crates.io-index)",
 "itoa 0.4.2 (registry+https://github.com/rust-lang/crates.io-index)",
 "log 0.4.3 (registry+https://github.com/rust-lang/crates.io-index)",
 "net2 0.2.33 (registry+https://github.com/rust-lang/crates.io-index)",
 "time 0.1.40 (registry+https://github.com/rust-lang/crates.io-index)",
 "tokio 0.1.7 (registry+https://github.com/rust-lang/crates.io-index)",
 "tokio-executor 0.1.2 (registry+https://github.com/rust-lang/crates.io-index)",
 "tokio-io 0.1.7 (registry+https://github.com/rust-lang/crates.io-index)",
 "tokio-reactor 0.1.2 (registry+https://github.com/rust-lang/crates.io-index)",
 "tokio-tcp 0.1.0 (registry+https://github.com/rust-lang/crates.io-index)",
 "tokio-timer 0.2.4 (registry+https://github.com/rust-lang/crates.io-index)",
 "want 0.0.6 (registry+https://github.com/rust-lang/crates.io-index)",
]
```

**Comment by seanmonstar** [comment]
Yikes, well alright, I filed https://github.com/carllerche/h2/issues/300 to look into it.

**Comment by seanmonstar** [comment]
This was fixed upstream, and released as h2 v0.1.12. A `cargo update -p h2` should fix this up.
